Sources/Build/BuildDelegate.swift

```diff
@@ -684,3 +689,66 @@ private extension Diagnostic.Message {
         .error("failed parsing the Swift compiler output: \(error)")
     }
 }
+
+#if canImport(CryptoKit)
+import CryptoKit
+
+/// A filesystem for LLBuild that returns hash of the files
+/// instead of stat information. This is obviously a hack
+/// until llbuild has proper support for content-based tracking.
+fileprivate struct LLBContentHashFileSystem: SPMLLBuild.FileSystem {
+    public struct HashFileInfo: SPMLLBuild.FileInfo {
+        var statBuf: stat
+
+        init(_ statBuf: stat) {
+            self.statBuf = statBuf
+        }
+    }
+
+    func fileContents(_ path: String) throws -> ByteString {
+        try localFileSystem.readFileContents(AbsolutePath(path))
+    }
+
+    func read(_ path: String) throws -> [UInt8] {
+        try fileContents(path).contents
+    }
+
+    func getFileInfo(_ path: String) throws -> SPMLLBuild.FileInfo {
+        if #available(macOS 10.15, *) {
+            var hasher = CryptoKit.SHA256()
+            hasher.update(data: try read(path))
```

Comment:
Are there any extended attributes that should be considered part of a file's contents?  I don't believe so (especially not since they wouldn't survive Git), but it might be worth considering.